### ISSUE TITLE
InfoWindow problems

### ISSUE THREAD

**Issue by pjsinco** [MicheleBertoli/react-gmaps#40]
Hi Michele.

I am having trouble with InfoWindow component—they are always open by default.

I dynamically populate my map with up to 25 Marker and InfoWindow components. I would like to have an InfoWindow open only when a marker is clicked. Will you please suggest the best way to do this?

If it helps, here is a snippet of my component. But please feel free to ignore it's not helpful.

```javascript
  handleMarkerClick: function(evt) {
    console.log('need to open the corresponding infowindow');
  },

  render: function() {
    function makeMarker(doctor, i) {
      return (
        <Marker
          key={i}
          lat={doctor.lat}
          lng={doctor.lng}
          onClick={this.handleMarkerClick}
        />
      )
    }

    function makeInfoWindow(doctor, i) {
      return (
        <InfoWindow
          key={i}
          lat={doctor.lat}
          lng={doctor.lng}
          ref={'info-' + doctor.id}
          content={doctor.full_name}
          onCloseClick={this.handleInfoWindowCloseClick}
        />
      )
    }

    const mapData = this.props.map
    const markers = mapData.markers.map(makeMarker, this)
    const infoWindows = mapData.markers.map(makeInfoWindow, this)

    return (
      <div className="sideBar_ElementHolder sideBar_MapElementHolder">
        <div className="sideBar_GoogleMapHolder">
          <div id="googleMapWrapper" className="google-map">
            <Gmaps
              width={'100%'}
              ref="gmaps"
              height={'325px'}
              lat={mapData.options.lat}
              lng={mapData.options.lng}
              onMapCreated={this.onMapCreated}
              loadingMessage={'hiyafromloadingmessage'}
              params={{ v: '3' }}
            >
              {markers}
              {infoWindows}
            </Gmaps>
          </div>
        </div>
      </div>
    )
  }
```

**Comment by MicheleBertoli** [comment]
Hello @pjsinco,

Starting from this points:
- Google Maps Info Windows are open by default
- React Components re-render when the state changes
- Creating the Info Windows on-demand is better

I created the following demo ([source](https://gist.github.com/MicheleBertoli/cdd3960f608574e49e24)):

![info](https://cloud.githubusercontent.com/assets/1308971/13546675/69a29370-e2ad-11e5-911f-94e1f20ec0b4.gif)

Basically, you have to keep track of the state of the Info Windows and show the open ones when the Component re-renders.

Please upgrade `react-gmaps` to the latest version ([1.4.3](https://github.com/MicheleBertoli/react-gmaps/releases/tag/1.4.3)).

I hope this answers your question, have a nice day!

**Comment by pjsinco** [comment]
Perfecto! Thank you for taking time to make the example. It is very helpful and much appreciated!

**Comment by prashanthravi19** [comment]
is there a way to set the code in ts?
